### PR TITLE
Refine assert IPEX

### PR DESCRIPTION
**Pull request by EikanWang**
TORCH_INTERNAL_ASSERT is not only for debug but also for release. I replaced it by TORCH_INTERNAL_ASSERT_DEBUG_ONLY. TORCH_INTERNAL_ASSERT_DEBUG_ONLY is only enabled in debug.

### CODE REVIEW

**Comment by EikanWang** [comment]
@jiayisunx @pinzhenx @hongzhen1 @XiaobingSuper

**Comment by EikanWang** [comment]
@jiayisunx @pinzhenx @hongzhen1 @XiaobingSuper

File: torch_ipex/csrc/cpu/ExtendOPs.cpp [hunk 1]
```diff
@@ -305,11 +305,11 @@ inline std::vector<at::Tensor> _interaction_backward(const at::Tensor & grad_out
 
 at::Tensor AtenIpexTypeExt::interaction_forward(const std::vector<at::Tensor> & input) {
   if (input[0].scalar_type() == at::kFloat) {
-    for (const auto &in : input) { TORCH_INTERNAL_ASSERT(in.scalar_type() == at::kFloat); }
```
Comment (EikanWang) [inline torch_ipex/csrc/cpu/ExtendOPs.cpp:308]: @hongzhen1 , Do you think it should be replaced by TORCH_CHECK?

Comment (hongzhen1) [reply torch_ipex/csrc/cpu/ExtendOPs.cpp:308]: it's OK for me

**Pull request merged**